src/check_datapackage/extensions.py

```diff
@@ -112,22 +158,20 @@ class RequiredCheck(BaseModel, frozen=True):
 
     jsonpath: JsonPath
     message: str
-    _field_name: str = PrivateAttr()
+    _targets: list[TargetJsonPath] = PrivateAttr()
 
     @model_validator(mode="after")
     def _check_field_name_in_jsonpath(self) -> Self:
-        field_name_match = re.search(r"(?<!\.)(\.\w+)$", self.jsonpath)
-        if not field_name_match:
-            raise ValueError(
-                f"Cannot use `RequiredCheck` for this JSON path `{self.jsonpath}`."
-                " A `RequiredCheck` must use a JSON path that targets a specific and"
-                " real property (e.g., `$.title`) or set of properties (e.g.,"
-                " `$.resources[*].title`). Unspecific JSON paths (e.g., `$..title`)"
-                " or JSON paths pointing to array items (e.g., `$.resources[0]`) are"
-                " not allowed."
-            )
-
-        object.__setattr__(self, "_field_name", field_name_match.group(1))
+        jsonpath = compile(self.jsonpath)
+        if isinstance(jsonpath, JSONPath):
+            paths = [jsonpath]
+        else:
+            first_path = cast(JSONPath, jsonpath.path)
+            paths = [first_path] + _map(jsonpath.paths, lambda path: path[1])
```

Comment:
```suggestion
            paths = [first_path] + _map(jsonpath.paths, itemgetter(0))
```

Rather than use lambda. Needs to import `from operator import itemgetter`